tests/util/setup_tests.rs

```diff
@@ -176,3 +186,45 @@ pub fn test_pool_rename(paths: &[&Path]) {
     assert_eq!(pool_name, name2);
     engine.teardown().unwrap();
 }
+
+/// Verify that setting up a pool when the pool has not been previously torn
+/// down does not fail. Clutter the orignally set up pool with a filesystem
```

Comment:
Looks very good to me, especially this test. 👍 from me once temp stuff is removed, and fix a typo in the above comment line.